File: 203-Remove-Linked-List-Elements.py
class Solution:
    def removeElements(self, head: Optional[ListNode], val: int) -> Optional[ListNode]:
        if not head:
            return None

        dummy = ListNode(val=0, next=head)
        prev, curr = dummy, head

        while curr:
            if curr.val == val:
                prev.next = curr.next
            else:
                prev = curr
            curr = curr.next

        return dummy.next

File: neetcode/203-Remove-Linked-List-Elements.ts
function removeElements(head: ListNode | null, val: number): ListNode | null {
  let dummy = new ListNode(0, null)
  dummy.next = head
  let curr = dummy

  while (curr) {
    while (curr.next && curr.next.val === val) {
      curr.next = curr.next.next
    }

    curr = curr.next
  }

  return dummy.next
};